src/printer.js

```diff
@@ -3127,12 +3143,16 @@ function printJSXElement(path, options, print) {
   let groups = [[]]; // Initialize the first line's group
   children.forEach((child, i) => {
     // leading and trailing JSX whitespace don't go into a group
-    if (child === jsxWhitespace) {
+    if (child === innerJsxWhitespace) {
       if (i === 0) {
-        groups.unshift(child);
+        if (children.length === 1) {
```

Comment:
I think it might be a bit more clear if this were pulled out... 

```js
if (child === innerJsxWhitespace) {
  if (children.length === 1) {
    groups.unshift(solitaryJsxWhitespace);
    groups.pop();  // remove unnecessary empty group
    return;
  } else if (i === 0) {
    groups.unshift(leadingJsxWhitespace);
    return;
  } else if (i === children.length - 1) {
    groups.push(trailingJsxWhitespace);
    return;
 }
```